### ISSUE TITLE
Deserializing missing properties to None

### ISSUE THREAD

**Issue by lucasfelber** [neo4j-labs/neo4rs#147]
Since null represents a missing value in neo4j I think that when deserializing a node to a struct it should fill any missing properties with None if it is defined as an Option.

``` rust
#[derive(Deserialize)]
struct Foo {
    bar: Option<String>
}

// the Foo node does not have any properties
let mut row = graph.execute(
    query("MATCH (foo:Foo) RETRUN foo")
).await?.next().await?;

if let Some(r) = row {
    r.get("foo")?.to::<Foo>().unwrap();
}
```

This code failes with `Other("deserializing additional data requires a struct")` but I think it should work with Foo.bar being None.

**Comment by madmikeross** [comment]
I would like to second @lucasfelber. Here is a minimal test to replicate the issue I'm observing.

```toml
[package]
name = "test-stuct-with-option"
version = "0.1.0"
edition = "2021"

serde = { version = "1", features = ["derive"] }
serde_json = "1.0"
neo4rs = "0.7.0-rc.3"
```

```rust
#[cfg(test)]
mod tests {
    use std::sync::Arc;
    use neo4rs::{Graph, query};
    use serde::Deserialize;

    #[derive(Deserialize)]
    struct StructWithOption {
        a: Option<String>
    }

    #[tokio::test]
    async fn should_read_struct_with_option() {
        let graph = Arc::new(Graph::new("bolt://localhost:7687", "neo4j", "neo4j").await.unwrap());
        let a_val: Option<String> = None;
        let mut result = graph.execute(query("CREATE (ts:TestStruct {a: $a}) RETURN ts")
            .param("a", a_val))
            .await
            .unwrap();
        let Ok(Some(row)) = result.next().await else { todo!() };
        let test_struct: StructWithOption = row.get("ts").unwrap();

        assert_eq!(test_struct.a, None);
    }
}
```
    
This test will fail with the following error:
```bash
called `Result::unwrap()` on an `Err` value: Other("deserializing additional data requires a struct")
thread 'database::tests::should_write_test_struct_to_neo4j' panicked at 'called `Result::unwrap()` on an `Err` value: Other("deserializing additional data requires a struct")', src/database.rs:362:53
stack backtrace:
```

**Comment by tausifcreates** [comment]
+1 for this.

**Comment by knutwalker** [comment]
Just released version 0.7.1 with a fix for this. Note that this only works if the optionality is encoded with `Option<T>`. With the current implementation, I couldn't make `#[serde(default = …)]` work in the same way.
